src/main/webapp/app/controllers/settingsController.js

```diff
@@ -251,7 +251,9 @@ vireo.controller("SettingsController", function ($controller, $injector, $scope,
                 };
 
                 $scope.resetConfiguration = function (type, name) {
-                    return $scope.settings.configurable[type][name].reset();
+                    if ($scope.settings.configurable[type][name].id) {
```

Comment:
@kaladay I chose to check null on the id, because that is the cause of the error. The object itself is not ever null. It's only the id that is null for the default value. Would it be better to set an id for the default value instead?

Reply:
Should the client application be allowed to identify whose settings are whose?

Reply:
The default object has no id. When a color is changed from default, the new value is stored in the MANAGED_CONFIGURATION table and given an id. When the reset function is called, it queries the id from the MANAGED_CONFIGURATION table and deletes the data from the table. If the reset function is called on the default value, there is no data in the MANAGED_CONFIGURATION table. This causes the null id exception.

My solution prevents the query when the current value is set to default.